File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/clone/CodeBaseCloner.java
package io.sch.historyscan.domain.contexts.codebase.clone;

import io.sch.historyscan.domain.hexagonalarchitecture.DDDService;

@DDDService
public record CodeBaseCloner(CodeBaseRepository codeBaseRepository) implements Clone {

    @Override
    public ClonedCodeBase from(CodeBaseToClone codeBaseToClone) throws CodeBaseAlreadyExistsException {
        return codeBaseRepository.clone(codeBaseToClone);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/clone/CodeBaseCloner.kt
package io.sch.historyscan.domain.contexts.codebase.clone

import io.sch.historyscan.domain.hexagonalarchitecture.DDDService

@DDDService
class CodeBaseCloner(private val codeBaseRepository: CodeBaseRepository) : Clone {
    @Throws(CodeBaseAlreadyExistsException::class)
    override fun from(codeBaseToClone: CodeBaseToClone?): ClonedCodeBase? {
        return codeBaseRepository.clone(codeBaseToClone)
    }
}
